sdk/python/kfp/_client.py

```diff
@@ -98,7 +98,8 @@ class Client(object):
   LOCAL_KFP_CONTEXT = os.path.expanduser('~/.config/kfp/context.json')
 
   # TODO: Wrap the configurations for different authentication methods.
-  def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client_id=None, other_client_secret=None, existing_token=None, cookies=None):
+  def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client_id=None, other_client_secret=None, existing_token=None):
+  def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client_id=None, other_client_secret=None, existing_token=None, cookies=None, proxy=None, ssl_ca_cert=None):
```

Comment:
drive-by comment:

It seems like the interface of kfp.Client() is bloated because of various types of connections/deployments we have. That makes me think automatically inferring the right behavior of the authentication will be more and more difficult. 

Should we be more explicit and ask the users to specify the type of connection here? That will make the behavior more predictable.

Reply:
the problem is that we don't know those cases before CE reports it, e.x. this proxy case.
The usage will be

client=kfp.Client(host=xxx, proxy=yyy,ssl_ca_cert=zzz) which seems OK.

For your concern, another solution (maybe better) is that we allow directly pass in configuration object. It doesn't solve the authentication (this also also is not), but it helps for advanced network connections.

Reply:
Sounds good. I am in favor of being more explicit in this case. Indeed it will be very hard to cover all possible cases on our end exhaustively.